HLint.hs

```diff
@@ -45,6 +47,22 @@ ignore "Unused LANGUAGE pragma"
 -- so they shouldn't be replaced with 'newtype' blindly
 ignore "Use newtype instead of data"
 
+
+----------------------------------------------------------------------------
+-- Hints with 'id' should use 'identity'
+----------------------------------------------------------------------------
+
+warn = any identity ==> or
+warn = all identity ==> and
+warn = (x >>= identity) ==> join x
+warn = (identity =<< x) ==> join x
+warn = mapM identity ==> sequence
+warn = mapM_ identity ==> sequence_
+warn = maybe x identity ==> fromMaybe x
+warn = mapMaybe identity ==> catMaybes
+warn = maybe Nothing identity ==> join
```

Comment:
I'll make a note to add some useful things to this list, I'm sure there are many more.